lib/iris/fileformats/grib/_load_convert.py

```diff
@@ -158,7 +158,9 @@ def _unscale(v, f):
 
     if isinstance(value, Iterable) or isinstance(factor, Iterable):
         def _masker(item):
-            result = ma.masked_equal(item, _MDI)
+            numerical_mdi = 2 ** 32 - 1
```

Comment:
add comment here describing the logic process here, and the lack of examples or integration tests at present